recipes/json-glib/all/conanfile.py

```diff
@@ -0,0 +1,126 @@
+from conans import ConanFile, tools, Meson, VisualStudioBuildEnvironment
+
+import os
+import glob
+import shutil
+
+required_conan_version = ">=1.29"
+
+class JsonGlibConan(ConanFile):
+    name = "json-glib"
+    description = "JSON-GLib implements a full JSON parser and generator using GLib and GObject, and integrates JSON with GLib data types."
+    topics = ("json", "glib")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://wiki.gnome.org/Projects/JsonGlib"
+    license = "GNU LGPL-2.1"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_introspection": ['enabled', 'auto', 'disabled'],
+        "gtk_doc": ['enabled', 'auto', 'disabled'],
+        "man": [True, False],
+        "tests": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_introspection": 'auto',
+        "gtk_doc": 'auto',
+        "man": False,
+        "tests": False,
+    }
+    generators = "pkg_config"
+    _meson = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def requirements(self):
```

Comment:
For cross building, you need to acess properties of the build system set by the build profile.
Don't lean on `tools.os_info.is_xxxx` or `platform` as these are not build profile dependent.
```suggestion
        return "build_subfolder"

    @property
    def _settings_build(self):
        return getattr(self, "settings_build", self.settings)

    def requirements(self):
```